From 02803381b0aabae08dca9d9f5ed4d771eb1ad18d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 17 Sep 2015 15:44:21 -0500
Subject: [PATCH 27/97] apply backport patch
 collateral-evolutions/network/0012-driver-quirks/drivers_net_wireless_cw1200_sdio.patch

---
 backports/drivers/net/wireless/cw1200/cw1200_sdio.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/backports/drivers/net/wireless/cw1200/cw1200_sdio.c b/backports/drivers/net/wireless/cw1200/cw1200_sdio.c
index d3acc85..6d75d51 100644
--- a/backports/drivers/net/wireless/cw1200/cw1200_sdio.c
+++ b/backports/drivers/net/wireless/cw1200/cw1200_sdio.c
@@ -253,6 +253,12 @@ static size_t cw1200_sdio_align_size(struct hwbus_priv *self, size_t size)
 	else
 		size = sdio_align_size(self->func, size);
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 2, 0))
+	/* A quirk to handle this was committed in 3.2-rc */
+	if (size == SDIO_BLOCK_SIZE)
+		size += SDIO_BLOCK_SIZE;  /* HW bug; force use of block mode */
+#endif
+
 	return size;
 }
 
-- 
2.5.1

